src/proxyfactory.rs

```diff
@@ -130,13 +129,57 @@ impl ProxyFactory {
                 socket_factory.clone(),
                 resolver,
                 local_workload_information,
+                false,
             );
             result.connection_manager = Some(cm);
             result.proxy = Some(Proxy::from_inputs(pi, drain).await?);
         }
 
         Ok(result)
     }
+
+    pub async fn create_ztunnel_inbound_listener(
```

Comment:
Can we add a comment here explaining that this is a listener that proxies traffic for the ztunnels own servers and not things that the ztunnel is proxying on behalf of? The "ztunnel inbound listener" seems a little overloaded/non-specific. Same with the logs. Maybe something like `ztunnel management listener` or something